raiden_contracts/tests/test_channel_unlock.py

```diff
@@ -0,0 +1,178 @@
+
+
+def test_merkle_root_0_items(
+        chain,
+        web3,
+        unlock_test,
+        secret_registry,
+        get_block,
+        create_contract,
+):
+    (locksroot, unlocked_amount) = unlock_test.call().getMerkleRootAndUnlockedAmountPublic(b'')
+    assert locksroot == EMPTY_MERKLE_ROOT
+    assert unlocked_amount == 0
+
+
+def test_merkle_root_1_item_unlockable(
+        chain,
+        web3,
+        get_accounts,
+        unlock_test,
+        secret_registry,
+        get_block,
+        create_contract,
+):
+    A = get_accounts(1)[0]
+    pending_transfers_tree = get_pending_transfers_tree(web3, [6])
+
+    secret_registry.transact({'from': A}).registerSecret(
+        pending_transfers_tree.unlockable[0][3]
+    )
+    assert secret_registry.call().getSecretRevealBlockHeight(
+        pending_transfers_tree.unlockable[0][2]
+    ) == web3.eth.blockNumber
+
+    (locksroot, unlocked_amount) = unlock_test.call().getMerkleRootAndUnlockedAmountPublic(
+        pending_transfers_tree.packed_transfers
+    )
+
+    merkle_root = get_merkle_root(pending_transfers_tree.merkle_tree)
+    assert locksroot == merkle_root
+    assert unlocked_amount == 6
+
+
+def test_merkle_root(
+        chain,
+        web3,
+        get_accounts,
+        unlock_test,
+        secret_registry,
+        get_block,
+        create_contract,
+):
+    (A, B) = get_accounts(2)
+    pending_transfers_tree = get_pending_transfers_tree(web3, [1, 3, 5], [2, 8, 3])
+
+    for lock in pending_transfers_tree.unlockable:
+        secret_registry.transact({'from': A}).registerSecret(lock[3])
+        assert secret_registry.call().getSecretRevealBlockHeight(lock[2]) == web3.eth.blockNumber
+
+    (locksroot, unlocked_amount) = unlock_test.call().getMerkleRootAndUnlockedAmountPublic(
+        pending_transfers_tree.packed_transfers
+    )
+    merkle_root = get_merkle_root(pending_transfers_tree.merkle_tree)
+
+    assert locksroot == merkle_root
+    assert unlocked_amount == 9
+
+
+def test_channel_unlock(
+        web3,
+        custom_token,
+        token_network,
+        secret_registry,
+        create_channel,
+        channel_deposit,
+        get_accounts,
+        create_balance_proof,
+        event_handler
+):
+    (A, B) = get_accounts(2)
+    settle_timeout = 8
+    deposit_A = 20
+    deposit_B = 30
+    additional_hash = fake_hex(32, '02')
+    locksroot1 = fake_hex(32, '00')
+    locked_amount1 = 0
+
+    # Create channel and deposit
+    channel_identifier = create_channel(A, B, settle_timeout)
+    channel_deposit(channel_identifier, A, deposit_A)
+    channel_deposit(channel_identifier, B, deposit_B)
+
+    # Mock pending transfers data
+    pending_transfers_tree = get_pending_transfers_tree(web3, [1, 3, 5], [2, 4], settle_timeout)
+    locksroot_bytes = get_merkle_root(pending_transfers_tree.merkle_tree)
+    locksroot2 = '0x' + locksroot_bytes.hex()
+    locked_amount2 = reduce((lambda x, y: x + y[1]), pending_transfers_tree.transfers, 0)
```

Comment:
This should be its own helper function to calculated the locked amount. Would be easier to read, plus would help with abstraction as you use it in other places too. e.g [here](https://github.com/raiden-network/raiden-contracts/pull/39/files#diff-a0ec52e4288fe8c7b94e7d1fc9704dc6R86)